### PR TITLE
CLI commands: fix search path for locating ef.dll/ef.exe

### PR DESCRIPTION
**Pull request by natemcmaster**
dotnet-ef currently assumes AppContext.BaseDirectory is in the extracted nupkg when really it is inside the special .tools folder in the nuget cache.
```
Expected: <nupkg>/packages/Ms.EfCore.Tools.DotNet/<version>/lib/netcoreapp1.0
Actual  : <nupkg>/packages/.tools/Ms.EfCore.Tools.DotNet/<version>/lib/netcoreapp1.0
```

This updates dotnet-ef to use Assembly.Location instead of AppContext.BaseDirectory to find the location of installed nupkg.

cc @JunTaoLuo @bricelam

### CODE REVIEW

File: test/Tools.DotNet.FunctionalTests/Utilities/DotnetEf.cs [hunk 1]
```diff
@@ -51,7 +51,8 @@ private IEnumerable<string> BuildArgsImpl(string[] args)
                 "--depsfile", _depsJson,
                 "--additionalprobingpath", _nugetPackageDir,
                 s_dotnetEfPath,
-                "--verbose"
+                "--verbose",
+                "--configuration", System.Environment.GetEnvironmentVariable("Configuration") ?? "Debug"
```
Comment (JunTaoLuo) [inline test/Tools.DotNet.FunctionalTests/Utilities/DotnetEf.cs]: why was this added?

Comment (natemcmaster) [inline test/Tools.DotNet.FunctionalTests/Utilities/DotnetEf.cs]: Test also happens to fail in release mode.

Comment (JunTaoLuo) [inline test/Tools.DotNet.FunctionalTests/Utilities/DotnetEf.cs]: Although this fixes that, but would be nice to know why? Also, did you verify this fix with Templates?

Comment (natemcmaster) [inline test/Tools.DotNet.FunctionalTests/Utilities/DotnetEf.cs]: I believe this is essentially a test parallelization problem.

Comment (JunTaoLuo) [inline test/Tools.DotNet.FunctionalTests/Utilities/DotnetEf.cs]: Okay, as long as no functional issues here.

Comment (natemcmaster) [inline test/Tools.DotNet.FunctionalTests/Utilities/DotnetEf.cs]: Oh, and yes. I've verified this fixes the issue.

**Comment by JunTaoLuo** [comment]
LGTM

**Comment by bricelam** [comment]
:shipit:
